ironfish-zkp/src/circuits/util.rs

```diff
@@ -143,3 +144,35 @@ pub fn assert_valid_asset_generator<CS: bellperson::ConstraintSystem<blstrs::Sca
 
     Ok(())
 }
+
+pub trait Reader: Sized {
```

Comment:
Why is this trait needed exactly?

Also, I'd suggest a name like `FromBytes` or similar, to convey the message that this is a trait that is used to _construct_ structs.

Reply:
How else would you do it without a trait extension? These are externally owned types. If I try to define an implementation directly it gives me a compiler error.

Reply:
I see: you're trying to add some utility methods to the external structs, so that the rest of the code can be shorter (I missed this when I first read the code). It makes senses then. I would just not make this trait public, but rather change its visibility to `pub(crate)`.